lib/spack/llnl/string.py

```diff
@@ -41,6 +41,20 @@ def comma_and(sequence: List[str]) -> str:
     return comma_list(sequence, "and")
 
 
+def ordinal(number: int) -> str:
+    """Return the ordinal number representation of the provided number.
```

Comment:
I didn't really understand the comment until I looked at what it was doing and googled "ordinal" -- but an example helps:
```suggestion
    """Return the ordinal representation (1st, 2nd, 3rd, etc.) for the provided number.
```